src/Engine/ProtoCore/DSASM/Heap.cs

```diff
@@ -17,20 +17,51 @@ public class HeapElement
 
         protected Heap heap;
         private int allocated;
-        private StackValue[] items;
 
-        public int VisibleSize { get; set; }
+        private StackValue[] values;
+        public virtual IEnumerable<StackValue> Values
+        {
+            get 
+            { 
+                for (int i = 0; i < Count; ++i)
+                    yield return values[i]; 
```

Comment:
This doesn't clone the collection - that's ok, but lets make that clear in the doc comments.